HLTrigger/Timer/src/memory_usage.cc

```diff
@@ -0,0 +1,79 @@
+#include <iostream>
+#include <dlfcn.h>
+
+#include "memory_usage.h"
+
+// see <jemalloc/jemalloc.h>
+extern "C" {
+  typedef
+  int (*mallctl_t)(const char *name, void *oldp, size_t *oldlenp, void *newp, size_t newlen);
+}
+
+namespace {
+  bool initialise();
+  const uint64_t * initialise_thread_allocated_p();
+  const uint64_t * initialise_thread_deallocated_p();
+
+  const uint64_t zero = 0UL;
+  thread_local const volatile uint64_t * thread_allocated_p   = initialise_thread_allocated_p();
```

Comment:
I want to make sure that the read is not optimised out, because the compiler does not see anything that could modify the value pointed to.
Do you think `volatile` is not needed here ?